clang/lib/Sema/SemaExprObjC.cpp

```diff
@@ -3206,9 +3206,11 @@ ExprResult SemaObjC::BuildInstanceMessage(
     }
     if (!isDesignatedInitChain) {
       const ObjCMethodDecl *InitMethod = nullptr;
+      auto *CurMD = SemaRef.getCurMethodDecl();
+      if (!CurMD)
+        return ExprResult((Expr *)nullptr);
```

Comment:
Returning an `ExprResult` with a null expression is surprising to me. If the intent is to indicate an error, then it would be better to return `ExprError()`. However, I suspect in this case that we would be better off just asserting that `CurMD` is not null. That is consistent with the code as previously written and, if the assertion is ever violated, we'll have an example that is informative with regard to what the behavior should be in that case.

Reply:
Thank you @tahonermann for review and feedback. I have added an assert.